scripts/updatebreaches.js

```diff
@@ -0,0 +1,97 @@
+"use strict";
+
+const got = require("got");
+
+const AppConstants = require("../app-constants");
+const HIBP = require("../hibp");
+
+
+const FX_RS_RECORDS = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-workspace/collections/fxmonitor-breaches/records`;
+const FX_RS_COLLECTION = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-workspace/collections/fxmonitor-breaches`;
+const FX_RS_PREVIEW_COLLECTION = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-preview/collections/fxmonitor-breaches`;
+const FX_RS_USERNAME = AppConstants.FX_REMOTE_SETTINGS_USERNAME;
+const FX_RS_PASSWORD = AppConstants.FX_REMOTE_SETTINGS_PASSWORD;
+
+
+if (!FX_RS_USERNAME || !FX_RS_PASSWORD) {
+  console.error("updatebreaches requires FX_REMOTE_SETTINGS_USERNAME and FX_REMOTE_SETTINGS_PASSWORD.");
+  process.exitCode = 1;
+  return;
+}
+
+
+function whichBreachesAreNotInRemoteSettingsYet(remoteSettingsBreachesSet, hibpBreaches) {
```

Comment:
emotional nit: Overall this function feels a bit strange because the two lists it's diff'ing aren't expected to be of the same type, and it doesn't handle the sourcing of the data so technically it can be used to compare any two lists of breaches. I'd expect either for this function to source the data instead of accepting arguments, or for this code to be inline'd. Up to you!

Reply:
Ah that's a good point ... and I requested a similar refactor on another project to make a function always *return* the same data *type*, so I totally agree. Hmm ...

My main motivation here was to break things up into smaller functions with fewer logic branches (`if`, `for` statements, etc.). To get the same effect, I'll move the hibp breaches call above the call to this function, and only pass in the hibp breaches. So, this function will handle sourcing the data from Remote Settings *and* returning only the HIBP breaches that aren't already there.

Good catch!